extern/benchmark/CMakeLists.txt

```diff
@@ -22,6 +23,12 @@ FetchContent_Declare( Benchmark ${DETRAY_BENCHMARK_SOURCE} )
 set( BENCHMARK_ENABLE_TESTING OFF CACHE BOOL
    "Turn off the tests in Google Benchmark" )
 
+# Respect specific expressions for IntelLLVM
+if ( "${CMAKE_CXX_COMPILER_ID}" MATCHES "IntelLLVM")
+   detray_add_flag( CMAKE_CXX_FLAGS "-fhonor-infinities")
+   detray_add_flag( CMAKE_CXX_FLAGS "-fhonor-nans")
+endif()   
```

Comment:
Warnings in the build of the externals are annoying, I fully agree. But I don't think we should be in the business of providing flags ourselves to the build of these externals.

The `-Werror` flag that we use for our own build should not reach the build of the external code. (If it does, that needs to be fixed.) So I would just not tweak the build of the Google externals, and just make my peace with seeing some warnings using Intel's compiler for now. We may just want to open a PR into the Google repositories to silence these warnings. But until them, let's just live with them in our own build.

Reply:
Right. There was a new option in google-benchmark (v1.6.1) which can suppress the Werror in release build so I added it